bundles/org.openhab.binding.ferroamp/src/main/java/org/openhab/binding/ferroamp/internal/FerroampHandler.java

```diff
@@ -0,0 +1,247 @@
+        super(thing);
+    }
+
+    @Override
+    public void handleCommand(ChannelUID channelUID, Command command) {
+        String transId = UUID.randomUUID().toString();
+        String valueConfiguration = command.toString();
+        if (FerroampBindingConstants.CHANNEL_REQUESTCHARGE.equals(channelUID.getId())) {
+            String requestCmdJsonCharge = "{\"" + "transId" + "\":\"" + transId
+                    + "\",\"cmd\":{\"name\":\"charge\",\"arg\":\"" + valueConfiguration + "\"}}";
+            FerroampMqttCommunication.sendPublishedTopic(requestCmdJsonCharge, ferroampConfig);
+        }
+        if (FerroampBindingConstants.CHANNEL_REQUESTDISCHARGE.equals(channelUID.getId())) {
+            String requestCmdJsonDisCharge = "{\"" + "transId" + "\":\"" + transId
+                    + "\",\"cmd\":{\"name\":\"discharge\",\"arg\":\"" + valueConfiguration + "\"}}";
+            FerroampMqttCommunication.sendPublishedTopic(requestCmdJsonDisCharge, ferroampConfig);
+        }
+        if (FerroampBindingConstants.CHANNEL_AUTO.equals(channelUID.getId())) {
+            String requestCmdJsonAuto = "{\"" + "transId" + "\":\"" + transId + "\",\"cmd\":{\"name\":\"auto\"}}";
+            FerroampMqttCommunication.sendPublishedTopic(requestCmdJsonAuto, ferroampConfig);
+        }
+    }
+
+    @Override
+    public void initialize() {
+        // Set channel configuration parameters
+        channelConfigEhub = FerroampChannelConfiguration.getChannelConfigurationEhub();
+        channelConfigSsoS1 = FerroampChannelConfiguration.getChannelConfigurationSsoS1();
+        channelConfigSsoS2 = FerroampChannelConfiguration.getChannelConfigurationSsoS2();
+        channelConfigSsoS3 = FerroampChannelConfiguration.getChannelConfigurationSsoS3();
+        channelConfigSsoS4 = FerroampChannelConfiguration.getChannelConfigurationSsoS4();
+        channelConfigEso = FerroampChannelConfiguration.getChannelConfigurationEso();
+        channelConfigEsm = FerroampChannelConfiguration.getChannelConfigurationEsm();
+
+        final MqttBrokerConnection ferroampConnection = new MqttBrokerConnection(ferroampConfig.hostName,
+                FerroampBindingConstants.BROKER_PORT, false, false, ferroampConfig.userName);
+
+        scheduler.scheduleWithFixedDelay(this::pollTask, 60, refreshInterval, TimeUnit.SECONDS);
+        this.setFerroampConnection(ferroampConnection);
+    }
+
+    private void pollTask() {
+        try {
+            startMqttConnection();
+        } catch (InterruptedException e) {
+            logger.debug("Problems with startMqttConnection()");
+        }
+
+        Objects.requireNonNull(ferroampConnection, "MqttBrokerConnection ferroampConnection cannot be null");
+        if (ferroampConnection.connectionState().toString().equals("DISCONNECTED")) {
+            updateStatus(ThingStatus.OFFLINE, ThingStatusDetail.COMMUNICATION_ERROR);
+            logger.debug("Problem connection to MqttBroker");
```

Comment:
Text + exit immediate
```suggestion
            logger.debug("Not connected to the MqttBroker");
            return
```

Reply:
20241125
Changes done as described

Reply:
Another fix that was reverted. Please re-apply.